test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5981,8 +5982,9 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
         c = None
         # 1- Submission using the command line tools
         if self.get_op_mode() == PTL_CLI:
+            runcmd = []
             exclude_attrs = []  # list of attributes to not convert to CLI
-            if isinstance(obj, Job):
+            if isinstance(obj, Job) and not qsub_from_script:
```

Comment:
Hi Bhroam,
I did try out putting env on command line. But since we are setting special characters like \t, \r in this test, it starts to error out. That is the reason i went with using script file to set environment variable.

Reply:
If the special characters are quoted properly, it should work.  If you can't get environmental variables on the command line to work, I still think the env dictionary is the right way to go.  You just have submit use du.create_temp_file() on the correct host and create the shell script that is used to submit the job.

Reply:
Thanks Broam, looks like i was not quoting the special characters properly. I will make the changes in submit.

Reply:
Had to use script file to set environment variable. Reason being we cannot set a function as environment variable due to shell shock malware fix